### PR TITLE
Adding dotgpg environment action

### PR DESCRIPTION
**Pull request by simonlevy5**
Action to support loading dotgpg encrypted variables into the ruby ENV.
This has two gem dependencies: ‘dotgpg’ and ‘dotgpg-environment’

### CODE REVIEW

**Comment by KrauseFx** [comment]
You can run `rubocop -a` to automatically fix the styling issues :+1:

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "Reads in production secrets set in a dotgpg file and puts them in ENV. More information at https://github.com/ConradIrwin/dotgpg"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :dotgpg_file,
+                                       env_name: "DOTGPG_FILE",
+                                       description: "Path to your DSYM file",
+                                       default_value: "dotgpg/secrets.gpg",
+                                       optional: true,
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: This value is not optional

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      def self.description
+        "Reads in production secrets set in a dotgpg file and puts them in ENV. More information at https://github.com/ConradIrwin/dotgpg"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :dotgpg_file,
+                                       env_name: "DOTGPG_FILE",
+                                       description: "Path to your DSYM file",
+                                       default_value: "dotgpg/secrets.gpg",
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: Change that to

```ruby
Dir["dotgpg/*.gpg"].last
```

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                                       end)
+        ]
+      end
+
+      def self.authors
+        ["simonlevy5"]
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac].include? platform
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: You can change that to `true`, as it works on all platforms.

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        end
+
+        raise "Dotgpg file '#{File.expand_path(dotgpg_file)}' not found".red if dotgpg_file && !File.exist?(dotgpg_file)
+
+        Helper.log.info "Reading secrets from #{dotgpg_file}"
+        Dotgpg::Environment.new(dotgpg_file).apply
+      end
+
+      def self.description
+        "Reads in production secrets set in a dotgpg file and puts them in ENV. More information at https://github.com/ConradIrwin/dotgpg"
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: Please remove the text after the `.`. More information can be in a separate method named `def self.details`

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    class DotgpgEnvironmentAction < Action
+      def self.run(options)
+        require 'dotgpg/environment'
+
+        if options[:dotgpg_file]
+          dotgpg_file = options[:dotgpg_file]
+        end
+
+        raise "Dotgpg file '#{File.expand_path(dotgpg_file)}' not found".red if dotgpg_file && !File.exist?(dotgpg_file)
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: The file validation should be handled in the `verify_block`, take a look at the other actions.

File: lib/fastlane/actions/dotgpg_environment.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  module Actions
+    module SharedValues
+    end
+
+    class DotgpgEnvironmentAction < Action
+      def self.run(options)
+        require 'dotgpg/environment'
+
+        if options[:dotgpg_file]
+          dotgpg_file = options[:dotgpg_file]
```
Comment (KrauseFx) [inline lib/fastlane/actions/dotgpg_environment.rb]: You can remove those 3 lines.

**Comment by KrauseFx** [comment]
Hey, thanks for the pull request! Sorry for adding so many comments. If you prefer, I can take over and do the changes for you.

**Comment by simonlevy5** [comment]
Thanks for the quick review! Feel free to fix the validator, it seems I used an out of date action as an example :grimacing: (hockey.rb has a validator in the run method)

**Comment by KrauseFx** [comment]
Thank you, I did some changes and merged it :rocket:

**Comment by KrauseFx** [comment]
Thanks again for your contribution :rocket:

**Pull request merged**